gcovr/coveralls_generator.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding:utf-8 -*-
+
+# This file is part of gcovr <http://gcovr.com/>.
+#
+# Copyright 2013-2019 the gcovr authors
+# Copyright 2013 Sandia Corporation
+# This software is distributed under the BSD license.
+# Filename: coveralls_generator.py
+# Author: Zachary J. Fields
+# Description: Module to generate Coveralls.io formatted JSON
+#              from coverage data collected by `gcovr`
+# Modification History:
+#
+
+from __future__ import absolute_import
+
+import json
+from datetime import datetime
+from functools import partial
+from .gitrepo.gitrepo import gitrepo
+from hashlib import md5
+from os import environ
+from .utils import presentable_filename
+
+
+def print_coveralls_report(covdata, output_file, options):
+    """
+    Outputs a JSON report in the Coveralls API coverage format
+
+    @param covdata: is a dictionary of file coverage objects, keyed with an absolute filepath
+    @param output_file: is the name of the file to create
+    @param options: options object
+    """
+
+    # Create object to collect coverage data
+    root = {}
+
+    # Capture timestamp
+    root['run_at'] = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S UTC")
+
+    # Pull environment variables
+    root['repo_token'] = environ.get('COVERALLS_REPO_TOKEN')
+
+    # Consume Travis CI specific environment variables _(if available)_
+    root['service_job_id'] = environ.get('TRAVIS_JOB_ID')
+    if (root['service_job_id'] is not None):
+        root['service_name'] = "travis-ci"
+        root['service_number'] = environ.get('TRAVIS_BUILD_NUMBER')
+        root['service_pull_request'] = environ.get('TRAVIS_PULL_REQUEST')
+    else:
+        root['service_name'] = ""
+        del root['service_job_id']
+
+    # Add last git commit information
+    root['git'] = gitrepo(options.root_dir)
```

Comment:
Is coverall tied to GIT?
In Jenkins this is also available as environment, I guess in Travis also. If you run the command localy you don't realy need the commit and therefore the dependency can be dropped.